special-pages/pages/duckplayer/app/features/replace-watch-links.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @typedef {import("./iframe").IframeFeature} IframeFeature
+ */
+
+import { VideoParams } from 'injected/src/features/duckplayer/util';
+
+export const WATCH_LINK_CLICK_EVENT = 'ddg-iframe-watch-link-click';
+
+/**
+ * @implements IframeFeature
+ */
+export class ReplaceWatchLinks {
+    /**
+     * @param {string} videoId
+     */
+    constructor(videoId) {
+        this.videoId = videoId;
+    }
+    /**
+     * @param {HTMLIFrameElement} iframe
+     */
+    iframeDidLoad(iframe) {
+        const doc = iframe.contentDocument;
+        const win = iframe.contentWindow;
+
+        if (!doc) {
+            console.log('could not access contentDocument');
+            return () => {};
+        }
+
+        if (win && doc) {
+            doc.addEventListener(
+                'click',
+                (e) => {
+                    /** @type {HTMLLinkElement|null} */
+                    const closestLink = /** @type {Element} */ (e.target).closest('a[href]');
```

Comment:
I wouldn’t cast in this scenario, just because it’s not our environment, add something like this instead to be safer and please Typescript at the same time

```js
if (!(e.target instanceof Element)) return;
```